eng/pipelines/templates/jobs/cosmos-ci.yml

```diff
@@ -0,0 +1,434 @@
+parameters:
```

Comment:
We shouldn't ideally have custom ci jobs in `eng/pipelines` for just Cosmos alone.

Reply:
I'll second this. If the purpose of this is so Cosmos can pass different Java versions for build we should work on making the standard CI job more configurable. These additional CI jobs add more maintenance and potential for skew.

Reply:
I agree we should not create specific templates for a few services, this would only be in place until we get the Cosmos team to add a Java 21 pipeline excluding the Spark connectors. ANother thing I can do for now is move the contents of this into the Cosmos `ci.yml` instead of having it in the `eng/` folder as a template.

Reply:
I also created [this issue](https://github.com/Azure/azure-sdk-for-java/issues/38266) to track excluding Cosmos Spark connectors from their runs.